src/components/FilterHeaderCategories.js

```diff
@@ -0,0 +1,112 @@
+// @flow
+import React from "react";
+import { View, StyleSheet, TouchableHighlight, Image } from "react-native";
+import type { StyleObj } from "react-native/Libraries/StyleSheet/StyleSheetTypes";
+import {
+  interestButtonBgColor,
+  interestButtonTextColor,
+  filterShowMeTextColor,
+  categoriesFilterButtonBgColor
+} from "../constants/colors";
+import text from "../constants/text";
+import Text from "./Text";
+import CategoriesPills from "./CategoriesPills";
+import chevronRightImg from "../../assets/images/chevronRight.png";
+
+type Props = {
+  selectedCategories: Set<string>,
+  onFilterPress: Function
+};
+
+type CategoriesFilterButtonProps = {
+  style?: StyleObj,
+  onPress: Function
+};
+
+const CategoriesFilterButton = ({
+  style,
+  onPress
+}: CategoriesFilterButtonProps) => (
+  <TouchableHighlight
+    accessibilityTraits={["button"]}
+    accessibilityComponentType="button"
+    style={[styles.categoriesFilterButton, style]}
+    onPress={onPress}
+  >
+    <Image source={chevronRightImg} />
+  </TouchableHighlight>
+);
+
+CategoriesFilterButton.defaultProps = {
+  style: {}
+};
+
+const FilterHeaderCategories = ({ selectedCategories, onFilterPress }: Props) =>
+  selectedCategories.size === 0 ? (
+    <View style={styles.contentInterest}>
+      <Text type="h1" style={styles.filterTitle}>
+        {text.filterTitle}
+      </Text>
+      <View style={styles.interestButton}>
+        <Text type="h2" style={styles.interestButtonText}>
+          {text.filterByInterest}
+        </Text>
+        <CategoriesFilterButton onPress={onFilterPress} />
+      </View>
+    </View>
+  ) : (
+    <View style={styles.categoryPillsContainer}>
+      <CategoriesPills
+        style={styles.categoryPills}
+        selectedCategories={selectedCategories}
+      />
+      <CategoriesFilterButton
+        style={styles.categoriesFilterOverButton}
+        onPress={onFilterPress}
+      />
+    </View>
+  );
+
+const styles = StyleSheet.create({
+  contentInterest: {
+    alignItems: "center",
+    flexDirection: "row"
+  },
+  filterTitle: {
+    color: filterShowMeTextColor,
+    paddingTop: 5,
+    marginRight: 8
+  },
+  categoryPillsContainer: {
+    flexDirection: "row"
+  },
+  categoriesFilterButton: {
+    width: 38,
+    height: 40,
+    backgroundColor: categoriesFilterButtonBgColor,
+    justifyContent: "center",
+    alignItems: "center",
+    borderTopRightRadius: 4,
+    borderBottomRightRadius: 4
+  },
+  categoryPills: {
+    borderTopRightRadius: 0,
+    borderBottomRightRadius: 0
+  },
+  categoriesFilterOverButton: {},
+  interestButton: {
+    width: 145,
+    height: 40,
```

Comment:
We can't use fixed width and height here, because we need to allow users to scale the font. I think `minHeight` should work fine. And maybe a `width` is not needed at all?